README.md

```diff
@@ -0,0 +1,1251 @@
+        │   └── gaiad
+        └── upgrade-info.json
+```
+
+By using auto-download setup (Method II), node runners do not need to prepare the new binaries manually. When propose upgrade, the `--upgrade-info` should include a link of the new binary. The environmental variable `DAEMON_ALLOW_DOWNLOAD_BINARIES` is set to be true.  Upon the upgrade height, if cosmovisor cannot find the new binnary locally, cosmovisor will download the new binary according to the instruction in the `upgrade-info.json`(this `data/up-grade-info.json` file is generated upon upgrade) and put the new binary in `cosmovior/upgrade/Vega/bin` and switch to run the new binary.
+
+*Note:*
+- In general, autodownload is not recommended, especially when the validators are not able to confirm the links are secure and correct.
+- For Vega upgrade, gaia will upgrade its dependency on  Cosmos SDK v0.42 to Cosmos SDK v0.44, this will require [Cosmovisor v0.1](https://github.com/cosmos/cosmos-sdk/releases/tag/cosmovisor%2Fv0.1.0). Later versions of cosmovisor do not support Cosmos SDK v0.42 or earlier if the auto-download option is enabled.
+- by using cosmovisor v0.1 you might have [node hanging issue](https://github.com/cosmos/cosmos-sdk/issues/9875) when query a result of large output size. For example, `gaiad q gov proposals` will hang the node being queried, this issue will not appear for cosmovisor versions later than v0.1.
+
+#### method I: manually prepare the new binary
+##### method I: set cosmosvisor
+create the folder for cosmosvisor for val1 and val2, and put the old binary in `cosmovisor/genesis/bin`.
+```shell
+mkdir -p $VAL_1_CHAIN_DIR/cosmovisor/genesis/bin
+mkdir -p $VAL_2_CHAIN_DIR/cosmovisor/genesis/bin
+cp $(which gaiad) $VAL_1_CHAIN_DIR/cosmovisor/genesis/bin
+cp $(which gaiad) $VAL_2_CHAIN_DIR/cosmovisor/genesis/bin
+```
+
+Build the new gaia binary
+```shell
+cd gaia
+git checkout start-upgrade
+make install
+```
+Create the folders for the two nodes and put the upgrade gaia binary into `cosmovisor/upgrades/Vega/bin`:
+```shell
+mkdir -p $VAL_1_CHAIN_DIR/cosmovisor/upgrades/Vega/bin
+mkdir -p $VAL_2_CHAIN_DIR/cosmovisor/upgrades/Vega/bin
+cp $(which gaiad) $VAL_1_CHAIN_DIR/cosmovisor/upgrades/Vega/bin
+cp $(which gaiad) $VAL_2_CHAIN_DIR/cosmovisor/upgrades/Vega/bin
+```
+##### method I: start cosmovisor
+For val1:
+```shell
+export DAEMON_NAME=gaiad
+export DAEMON_HOME= $(pwd)/$VAL_1_CHAIN_DIR
+export DAEMON_RESTART_AFTER_UPGRADE=true
+cosmovisor start --x-crisis-skip-assert-invariants --home $VAL_1_CHAIN_DIR
+```
+For val2:
+
+open a new terminal:
+```shell
+export DAEMON_NAME=gaiad
+export DAEMON_HOME= $(pwd)/$VAL_2_CHAIN_DIR
+export DAEMON_RESTART_AFTER_UPGRADE=true
+cosmovisor start --x-crisis-skip-assert-invariants --home $VAL_2_CHAIN_DIR
+```
+##### Method I: propose upgrade
+The user owns by val2 is a delegator. So user can vote. Since we changed the gov parameters, the delegations this user delegated are far enough for this proposal to pass.
```

Comment:
```suggestion
The user owned by validator 2 is a delegator, so this user can vote. Since we changed the gov parameters, the voting power this user has delegated is more than enough for this proposal to pass.
```